crates/weaver_resolved_schema/src/catalog.rs

```diff
@@ -31,3 +31,13 @@ pub enum Stability {
     /// A stable definition.
     Stable,
 }
+
+impl Catalog {
+    /// Returns the attribute name from an attribute ref if it exists
+    /// in the catalog or None if it does not exist.
+    pub fn attribute_name(&self, attribute_ref: &AttributeRef) -> Option<String> {
```

Comment:
This is a bit of a rust design question.

I realize, as this is a command line tool, we're being a bit "free" with `clone` for now. However, when do you prefer `String` vs. `&str`?  This seems like an instance where just returning `&str` may be more ergonomic of an API.

Reply:
Good catch. In this context, the best practice is to return Option<&str> since the attribute name is owned by the catalog, and it's most likely that the catalog's lifetime will not be ephemeral on the caller's side. I will make this change. Additionally, I will check if there is a `Clippy` lint to enforce this best practice.

Reply:
I didn't find any Clippy lint to automatically check this.